### PR TITLE
bug fix to issue #1331

### LINKED ISSUE #1331: Setting `__TBB_DEFAULT_PARTITIONER` doesn't set partitioner in `parallel_for` with step
`__TBB_DEFAULT_PARTITIONER` is not used for `parallel_for` overload that utilizes step over a range, so even if I expect it to use `affinity_partitioner`, by setting `__TBB_DEFAULT_PARTITIONER` to `affinity_partitioner`, it still uses `auto_partitioner` (which is btw default value of `__TBB_DEFAULT_PARTITIONER`)

### PR DESCRIPTION
**Pull request by blonded04**
### Description 
Fix that functions that are supposed to use `__TBB_DEFAULT_PARTITIONER` are using `auto_partitioner`. 

Even though by default `__TBB_DEFAULT_PARTITIONER` is `auto_partitioner`: when setting `__TBB_DEFAULT_PARTITIONER` to, for example, `simple_partitioner`, `auto_partitioner` was used, not `simple_partitioner` (which would be used now which is good)

Fixes #1331 

- [x] - git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/oneapi-src/oneTBB/blob/master/CONTRIBUTING.md#pull-requests) for details)_

### Type of change

_Choose one or multiple, leave empty if none of the other choices apply_

_Add a respective label(s) to PR if you have permissions_

- [x] bug fix - _change that fixes an issue_
- [ ] new feature - _change that adds functionality_
- [ ] tests - _change in tests_
- [ ] infrastructure - _change in infrastructure and CI_
- [ ] documentation - _documentation update_

### Tests

- [ ] added - _required for new features and some bug fixes_
- [x] not needed

### Documentation

- [ ] updated in # - _add PR number_
- [ ] needs to be updated
- [x] not needed

### Breaks backward compatibility
- [ ] Yes
- [x] No
- [ ] Unknown

### Notify the following users
_List users with `@` to send notifications_

### CODE REVIEW

File: include/oneapi/tbb/parallel_for.h [hunk 1]
```diff
@@ -319,7 +319,7 @@ void parallel_for_impl(Index first, Index last, Index step, const Function& f, P
 template <typename Index, typename Function>
     __TBB_requires(parallel_for_index<Index> && parallel_for_function<Function, Index>)
 void parallel_for(Index first, Index last, Index step, const Function& f) {
-    parallel_for_impl<Index,Function,const auto_partitioner>(first, last, step, f, auto_partitioner());
+    parallel_for_impl<Index,Function,const __TBB_DEFAULT_PARTITIONER>(first, last, step, f, __TBB_DEFAULT_PARTITIONER());
```
Comment (blonded04) [inline include/oneapi/tbb/parallel_for.h:322]: is `const` redundant here? in case `__TBB_DEFAULT_PARTITIONER` is `affinity_partitioner` for example

Comment (blonded04) [reply include/oneapi/tbb/parallel_for.h:322]: nvm, it's ok for `affinity_partitioner` to be `const`, anyway `affinity_partitioner::task_partition_type` is used for actual partitioning

**Comment by blonded04** [comment]
@kboyarinov hello! I guess all the conditions from CONTRIBUTING.md are satisified and PR is safe to run in CI.

**Review by sarathnandu — APPROVED** [review]

**Review by dnmokhov — APPROVED** [review]
LGTM

**Pull request merged**